File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginActivityMVP.java
package com.charlye934.loginddagger.login;

public interface LoginActivityMVP {

    interface View{
        String getFirstName();
        String getLastName();

        void showUserNotAvaible();
        void showInputError();
        void showUserSaved();

        void setFirstName(String firstName);
        void setLasNmae(String lastName);
    }

    interface  Presenter{
        void setView(LoginActivityMVP.View view);
        void loginButtonClicked();
        void getCurrentUser();
    }

    interface Model{
        void createUser(String firstName, String lastName);
        User getUser();
    }
}

File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginActivityMVP.kt
package com.charlye934.loginddagger.login

interface LoginActivityMVP {

    interface View{
        var getFirstName:String
        var getLastName:String

        fun showUserNotAvaible()
        fun showInputError()
        fun showUserSaved()

        fun setFirstName(firstName: String)
        fun setLastName(lastName: String)
    }

    interface Presenter{
        fun setView(view:LoginActivityMVP.View)
        fun loginButtonClicked()
        fun getCurrentUser()

    }

    interface Model{
        fun createUser(firstName:String, lastName:String)
        fun getUser():User
    }
}